utils/ansible-runner-service.sh

```diff
@@ -415,36 +415,29 @@ check_access() {
 
 manage_ansible_hosts_file() {
     echo "Linking the runner service inventory to ceph-ansible hosts"
-    ln -sf /usr/share/ceph-ansible/hosts /usr/share/ansible-runner-service/inventory/hosts
-
-    echo "Linking the default ansible hosts file to the ceph-ansible hosts file."
-    if [ -h "$DEFAULT_ANSIBLE_HOSTS" ]; then
-        # hosts file is a symlink
-        if $VERBOSE; then echo "- default ansible hosts is already a symlink, checking further"; fi
-        tgt=$(readlink -f "$DEFAULT_ANSIBLE_HOSTS")
-        if [ "$tgt" == "/usr/share/ceph-ansible/hosts" ]; then
-            if $VERBOSE; then echo "- default ansible hosts already linked to ceph-ansible hosts. No changes required"; fi
-        else
-            echo "WARNING: Unable to link $DEFAULT_ANSIBLE_HOSTS to ceph-ansible hosts, it's already a symlink to '$tgt'"
-            echo "Admin intervention needed to reconcile the inventory files, or use -i on all ansible-playbook invocations"
-        fi
-    else
-        # hosts file is a regular file
-        hosts_dir=$(dirname "$DEFAULT_ANSIBLE_HOSTS")
-        save_file="${DEFAULT_ANSIBLE_HOSTS}.orig"
+    # create hosts file if it doesn't exist
+    [ ! -f $CEPH_ANSIBLE_HOSTS ] && touch $CEPH_ANSIBLE_HOSTS
+
+    # Set up a new hosts file if the hosts file contains hosts details already or it is an INI file
+    if grep -q -e 'hosts:' -e ']' "$CEPH_ANSIBLE_HOSTS" 2>/dev/null; then
```

Comment:
an inventory could just be a list of servers (not grouped), so perhaps just a -s test would suffice i.e. it exists and it has content.